wave_lang/kernel/wave/analysis/index_sequence_analysis.py

```diff
@@ -946,26 +949,110 @@ def apply_offset(node: fx.Node):
     trace.walk(apply_offset)
 
 
+def get_index(custom: CustomOp):
+    """Get the index from a CustomOp, handling special cases like MMABase."""
+    if not custom.indexing_dims:
+        return None
+    if isinstance(custom, MMABase):
+        return custom.acc.index
+    return custom.index
+
+
 def create_broadcast(
-    binary_op: BinaryPyOp,
-    to_broadcast: CustomOp,
+    op: CustomOp,
+    to_broadcast_arg_name: str,
     broadcast_dim: IndexSymbol,
     broadcast_size: int,
     target_node: CustomOp,
 ):
     """
-    Create a broadcast node for the given binary operator.
+    Create a broadcast node for any multi-operand operator.
     """
-    with binary_op.graph.inserting_before(binary_op.fx_node):
+    to_broadcast = get_custom(getattr(op, to_broadcast_arg_name))
```

Comment:
One final thing, can we use to_broadcast directly instead of trying to find it again with `get_custom(getattr(op, to_broadcast_arg_name))`. I was thinking we can change the call to `to_broadcast` with operand["custom"] instead of operand["id"], for e.g:
```
create_broadcast(
                    custom,
                    operand["custom"],
                    target["dim"],
                    target["size"],
                    target["custom"],
                )
```

Reply:
We do need the name for when we update later, IE `op.update_arg(to_broadcast_arg_name, custom.fx_node)`.  So I think I was trying to go for fewer arguments.  We could get the argument name from the custom op's position in the parent, but that seems harder.  I'll push a commit that adds another argument, and you can decide either way.